### DISCUSSION TITLE
Transaction not working

### DISCUSSION THREAD

**Discussion by eclipsexml** [linq2db/linq2db#3215] [Q&A] [answered]
I use DI in asp.net core but use transaction  not working as follows

```cs
 using (var scope = serviceProvider.CreateScope())
 {
      using (var testDataConnection = scope.ServiceProvider.GetService<DataConnection>())
      {
            try{
                testDataConnection.BeginTransaction();
                ... insert
                throw new Exception(123);
              }
              catch(Exception ex){
                  testDataConnection.RollbackTransaction()
              }
      }
}
```

If you have a question, first try to check our [documentation](https://linq2db.github.io/), especially [FAQ](https://linq2db.github.io/articles/FAQ.html) and search in [issues](https://github.com/linq2db/linq2db/issues) and [discussions](https://github.com/linq2db/linq2db/discussions) - maybe your question already answered there.

If you still have questions or want to start general discussion, please create new topic in [discussions](https://github.com/linq2db/linq2db/discussions).

If you are seeing an exception, include the full exceptions details (message and stack trace).

```
Exception message:
Stack trace:
```

### Steps to reproduce
Include a complete code listing (or project/solution) that we can run to reproduce the issue.

Partial code listings, or multiple fragments of code, will slow down our response or cause us to push the issue back to you to provide code to reproduce the issue.

```c#
<code with error and mapping classes>
```

### Environment details
linq2db version: *?*
Database Server: *?*
Database Provider: *?*
Operating system: *?*
.NET Framework: *?*

**Comment by MaceWindu** [comment]
What do you mean by "not working"? What exactly doesn't work as you expected?

**Reply by MaceWindu** [reply, marked as answer]
I would suspect you don't see your data in database (as there is no Commit call in your example)?

```cs
 using (var scope = serviceProvider.CreateScope())
 {
      using (var testDataConnection = scope.ServiceProvider.GetService<DataConnection>())
      using (var transaction = testDataConnection.BeginTransaction())
      {
                ... insert
                transaction.Commit();
      }
}
```
